app/controllers/testsets_controller.rb

```diff
@@ -121,12 +121,12 @@ def testset_params
       params.require(:testset).permit(:name, :description, :collection_id, :user_id)
     end
 
-    def prepare_errors
-      nil unless @testset && Array(@testset.errors).size > 0
+    def prepare_errors(err_instance)
```

Comment:
This clever business needs to be applied to TestActions, and ObjectIdentifierSiblings. It might need to be done on ObjectIdentifiers